pkg/validate/validate.go

```diff
@@ -176,3 +177,11 @@ func MAC(addr string) error {
 	_, err := net.ParseMAC(addr)
 	return err
 }
+
+// IPNotInCIDR validates that a given IP is not contained within a CIDR subnet
+func IPNotInCIDR(ip string, cidr ipnet.IPNet) error {
```

Comment:
Also openstack needs the VIPs to be in the machineCIDR 
https://github.com/openshift/installer/pull/1959/commits/ade37f550cc2b4f26f23d1705b6a34e020a0bd5f
cc @tomassedovic

Reply:
This is true. Same does go for baremetal, upon further investigation.